pkg/kinflate/tree/builder.go

```diff
@@ -64,30 +68,38 @@ func manifestToManifestData(m *manifest.Manifest) (*ManifestData, error) {
 }
 
 // loadManifestFileFromPath loads a manifest object from file.
-func loadManifestFileFromPath(filename string) (*manifest.Manifest, error) {
-	loader := kutil.ManifestLoader{fs.MakeRealFS()}
-	m, err := loader.Read(filename)
+func (l *Loader) loadManifestFileFromPath(filename string) (*manifest.Manifest, error) {
+	var loader = kutil.ManifestLoader{FS: l.FS}
+	fullFilename, err := loader.MakeValidManifestPath(filename)
+	if err != nil {
+		return nil, err
+	}
+	l.Path = fullFilename
```

Comment:
I'm a little bit confused by this variable. Who is supposed to set it? who is supposed to read it? in what order?

Reply:
Yes, this is confusing. We have a Loader abstraction which creates a ManifestLoader object. We intend to clean this up by integrating Loader and ManifestLoader into a coherent whole. But we don't want to do that in this initial PR. So basically, the ManifestLoader (called loader) calculated the full manifest file path. Then we store this value in the Loader object (l).